synapse/config/_base.py

```diff
@@ -908,8 +910,14 @@ def read_file(file_path: Any, config_path: Iterable[str]) -> str:
         os.stat(file_path)
         with open(file_path) as file_stream:
             return file_stream.read()
-    except OSError as e:
-        raise ConfigError("Error accessing file %r" % (file_path,), config_path) from e
+    except OSError:
+          try:
+            file_path = os.path.realpath(file_path)
+            os.stat(file_path)
+            with open(file_path) as file_stream:
+                return file_stream.read()
```

Comment:
I'm trying to understand in which circumstances `open(realpath(file_path))` would succeed where `open(file_path)` has failed. Please could you enlighten me?

Reply:
Yes.
If it was symlink, than `open()` will fail (because symlink can not be read this way), but `open( os.realpath())` will try to open file, symlink points to.
Espesialy usefull with certbot.

Reply:
I'm afraid I'm not following. `open()` can read symlinks just fine:

```
rav@fred:~/tmp/x$ ls -l
total 4
lrwxrwxrwx 1 rav rav 10 Jul 12 12:25 linkfile -> target.txt
-rw-rw-r-- 1 rav rav  5 Jul 12 12:25 target.txt
rav@fred:~/tmp/x$ cat target.txt 
1234
rav@fred:~/tmp/x$ python3
Python 3.8.5 (default, Jan 27 2021, 15:41:15) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> with open("linkfile") as stream:
...     print(stream.read())
... 
1234

```

Reply:
For me this wasnt true

Reply:
> For me this wasnt true

if that's the case, you might want to consider filing a bug against your python distribution.